code/game/objects/machinery/nuclearbomb.dm

```diff
@@ -205,125 +213,134 @@
-	Time: [has_auth ? "<a href='?src=[REF(src)];time=-10'>--</a> <a href='?src=[REF(src)];time=-1'>-</a> [timeleft] <a href='?src=[REF(src)];time=1'>+</a> <a href='?src=[REF(src)];time=10'>++</a>" : "- - [timeleft] + +"] <br />
-	<br />
-	Safety: [safety ? "On" : "Off"] [has_auth ? "<a href='?src=[REF(src)];safety=1'>Toggle</a>" : "Toggle"] <br />
-	Anchor: [anchored ? "Engaged" : "Off"] [has_auth ? "<a href='?src=[REF(src)];anchor=1'>Toggle</a>" : "Toggle"] <br />
-	<hr />
-	Red Auth. Disk: <a href='?src=[REF(src)];disk=red'>[r_auth ? "++++++++++" : "----------"]</a><br />
-	Green Auth. Disk: <a href='?src=[REF(src)];disk=green'>[g_auth ? "++++++++++" : "----------"]</a><br />
-	Blue Auth. Disk: <a href='?src=[REF(src)];disk=blue'>[b_auth ? "++++++++++" : "----------"]</a><br />
-	"}
-
-	var/datum/browser/popup = new(user, "nuclearbomb", "<div align='center'>Nuclear Bomb</div>", 300, 400)
-	popup.set_content(html)
-	popup.open()
-
-
-
-/obj/machinery/nuclearbomb/Topic(href, href_list)
+	if(timeleft <= 5)
+		status = "Initializing nuclear explosion. Have a nice day :)"
+
+	data["status"] = status
+
+	return data
+
+/obj/machinery/nuclearbomb/ui_act(action, list/params)
 	. = ..()
 	if(.)
 		return
+	switch(action)
+		if("toggle_timer")
+			toggle_timer()
+		if("change_time")
+			change_time(params["seconds"])
+		if("toggle_safety")
+			toggle_safety()
+		if("toggle_anchor")
+			toggle_anchor()
+		if("toggle_disk")
+			toggle_disk(params["disktype"])
+
+///Toggles the timer on or off
+/obj/machinery/nuclearbomb/proc/toggle_timer()
+	if(has_auth)
+		if(exploded)
+			return
+		if(safety)
+			to_chat(usr, span_warning("The safety is still on."))
+			return
+		if(!anchored)
+			to_chat(usr, span_warning("The anchors are not set."))
+			return
+
+		timer_enabled = !timer_enabled
+
+		if(timer_enabled)
+			start_processing()
+
+		if(!lighthack)
+			icon_state = (timer_enabled) ? "nuclearbomb2" : "nuclearbomb1"
+
+///Modifies the nuke timer
+/obj/machinery/nuclearbomb/proc/change_time(time)
+	if(has_auth && !timer_enabled)
+		timemax += time
+		timemax = clamp(timemax, initial(timemax), 600)
+		timeleft = timemax
 
-	if(href_list["disk"])
-		var/disk_colour = href_list["disk"]
-		var/disk_type
-		var/obj/item/disk/nuclear/disk_slot
+///Toggles the safety on or off
+/obj/machinery/nuclearbomb/proc/toggle_safety()
+	if(has_auth)
+		safety = !safety
+		if(safety)
+			timer_enabled = FALSE
+			stop_processing()
+
+///Toggles the anchor bolts on or off
+/obj/machinery/nuclearbomb/proc/toggle_anchor()
+	if(has_auth)
+		if(removal_stage == NUKE_STAGE_BOLTS_REMOVED)
+			anchored = FALSE
+			visible_message(span_warning("\The [src] makes a highly unpleasant crunching noise. It looks like the anchoring bolts have been cut."))
+			return
+		if(istype(get_area(loc), /area/shuttle))
+			to_chat(usr, span_warning("This doesn't look like a good spot to anchor the nuke."))
+			return
```

Comment:
more usr and tochat not going to repeat myself for the rest

Reply:
done